lib/tasks/dev.rake

```diff
@@ -0,0 +1,89 @@
+desc "Fill the database tables with some sample data"
+task({ :sample_data => :environment}) do
+  
+  User.destroy_all
+  Loan.destroy_all
+  Adjustment.destroy_all
+
+  # Create Sample Users
+    #  email                  :string           default(""), not null
+    #  encrypted_password     :string           default(""), not null
+  email = ["alice@example.com", "bob@example.com", "carol@example.com"]
+
+  3.times do |count|
+    user = User.new
+    user.email = email.at(count)
+    user.password = "password"
+    user.save
+  end
+
+  p "Added #{User.count} Users"
+
+  # Create Sample Loans
+    #  current_balance     :float
+    #  original_amount     :float
+    #  interest_rate       :float
+    #  periods_in_year     :integer
+    #  user_id             :integer
+    #  loan_name           :string
+    #  monthly_min_payment :float
+
+    #cur_bal = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+    org_bal = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+    int_rate = [2, 3, 4, 5, 6]
+    payback_per = [2, 4, 6]
+    per_in_year = 12
+    users = User.all
+    l_name = ["Car", "Car 2", "Student Loan", "Grad School", "Credit Card"]
+
+    10.times do
+      loan = Loan.new
+      loan.current_balance = org_bal.sample
+      loan.original_amount = loan.current_balance
+      loan.interest_rate = int_rate.sample
+      loan.periods_in_year = per_in_year
+      loan.user_id = users.sample.id
+      loan.loan_name = l_name.sample
+      
+      # variables for min payments
+      rate = loan.interest_rate/100/12
+      periods = payback_per.sample * loan.periods_in_year
+
+      loan.monthly_min_payment = loan.original_amount * ( rate*(1 + rate)**periods) / ((1+rate)**periods -1)
+
+      loan.save
```

Comment:
One thing that I do by habit is include something like:

```ruby
p loan.errors.full_messages if loan.errors.any?
```

after every `.save` in my sample data file. Then if there are any validation errors (which happens a lot when I am shaking things out) I get immediate visible feedback.

Reply:
That's a great call out, that will provide much more visibility, especially when I start updated a couple things in my models and databases.  Thanks!